graphql/src/schema/ast.rs

```diff
@@ -411,6 +412,23 @@ pub fn get_input_object_definitions(schema: &Document) -> Vec<InputObjectType> {
         .collect()
 }
 
+/// Determines if the field is an imported type
+pub fn is_field_imported(
+    subgraph_id: &SubgraphDeploymentId,
+    schema: &Document,
+    field_definition: &Field,
+) -> bool {
+    let field_type_name = get_field_name(&field_definition.field_type);
+    Schema::new(subgraph_id.clone(), schema.clone())
+        .imported_types()
+        .iter()
+        .find(|(imported, _)| match imported {
+            ImportedType::Name(name) => name.eq(&field_type_name),
+            ImportedType::NameAs(_, az) => az.eq(&field_type_name),
```

Comment:
If you add a `ImportedType::name(&self) -> &str`, this could just be `&field_type_name == imported.name()` which would make this code easier to read.